packages/synapse-react-client/src/components/SynapseTable/table-top/DownloadOptions.tsx

```diff
@@ -37,9 +41,13 @@ export const DownloadOptions: React.FunctionComponent<
   const { onDownloadFiles, darkTheme = true } = props
 
   const fileColumnId = getFileColumnModelId(queryResultBundle?.columnModels)
-  const showAddQueryToDownloadList =
-    fileColumnId ?? canTableQueryBeAddedToDownloadList(entity)
 
+  const isEntityViewWithoutFiles =
+    entity && isEntityView(entity) && !hasFilesInView(entity)
+
+  const showAddQueryToDownloadList =
+    !isEntityViewWithoutFiles &&
+    (fileColumnId || canTableQueryBeAddedToDownloadList(entity))
```

Comment:
If this is an EntityView without files, _never_ show "Add to download list". The likely ENTITY_ID column will never have a value that can be added to the download list.